ArduCopter/mode_autotune.cpp

```diff
@@ -1222,11 +1224,73 @@ void Copter::ModeAutoTune::twitching_test(float measurement, float target, float
     }
 }
 
-// updating_d_up - increase D and adjust P to optimize the D term for a little bounce back
+// twitching_test_angle - twitching tests
+// update min and max and test for end conditions
+void Copter::ModeAutoTune::twitching_test_angle(float angle, float rate, float angle_target_max, float &meas_angle_min, float &meas_angle_max, float &meas_rate_min, float &meas_rate_max)
+{
+    // capture maximum angle
+    if (angle > meas_angle_max) {
+        // the angle still increasing
+        meas_angle_max = angle;
+        meas_angle_min = angle;
+    }
+
+    // capture minimum angle after we have reached a reasonable maximum angle
+    if ((angle < meas_angle_min) && (meas_angle_max > angle_target_max * 0.5f)) {
+        // the measurement is bouncing back
+        meas_angle_min = angle;
+    }
+
+    // capture maximum rate
+    if (rate > meas_rate_max) {
+        // the measurement is still increasing
+        meas_rate_max = rate;
+        meas_rate_min = rate;
+    }
+
+    // capture minimum rate after we have reached maximum rate
+    if (rate < meas_rate_min) {
+        // the measurement is still decreasing
+        meas_rate_min = rate;
+    }
+
+    // calculate early stopping time based on the time it takes to get to 90%
+    if (meas_angle_max < angle_target_max * 0.75f) {
```

Comment:
Comments say 90% threshold, code appears to be checking 75%

Reply:
Thanks, will do.